fix: query operator $exists does not work correctly with tags (#911) (#923)
docarray/docarray: docs/fundamentals/documentarray/find.md

(find-documentarray)=

# Query by Conditions

> We can use {meth}`~docarray.array.mixins.find.FindMixin.find` to select Documents from a DocumentArray based the
> conditions specified in a `query` object. One can use `da.find(query)` to filter Documents and get nearest neighbours
> from `da`:

Reviewer: ```suggestion
You can use {meth}`~docarray.array.mixins.find.FindMixin.find` to select Documents from a DocumentArray based the
conditions specified in a `query` object. You can use `da.find(query)` to filter Documents and get nearest neighbors
from `da`:
```

- To filter Documents, the `query` object is a Python dictionary object that defines the filtering conditions using
  a [MongoDB](https://docs.mongodb.com/manual/reference/operator/query/)-like query language.
- To find nearest neighbours, the `query` object needs to be a NdArray-like, a Document, or a DocumentArray object that

>   defines embedding. One can also use `.match()` function for this purpose, and there is a minor interface difference
>   between these two functions, which will be described {ref}`in the next chapter<match-documentarray>`.

Reviewer: ```suggestion
  defines embedding. You can also use `.match()` function for this purpose, and there is a minor interface difference
  between these two functions, which is described {ref}`in the next chapter<match-documentarray>`.
```

{ <field1>: { <operator1>: <value1> }, ... }
```

> Here `field1` is {ref}`any field name<doc-fields>` of a Document object. To access nested fields, one can use the dunder

Reviewer: ```suggestion
Here `field1` is {ref}`any field name<doc-fields>` of a Document object. To access nested fields, you can use the dunder
```

```

```json
[
  {
    "id": "4045a9659875fd1299e482d710753de3",
    "modality": "A",
    "tags": {
      "h": 14.0,
      "uom": "cm",
      "w": 21.0
    },
    "text": "journal",
    "weight": 25.0
  },
  {
    "id": "cf7691c445220b94b88ff116911bad24",
    "modality": "D",
    "tags": {
      "h": 22.85,
      "uom": "cm",
      "w": 30.0
    },
    "text": "planner",
    "weight": 75.0
  }
]
```

> Beside using a predefined value, one can also use a substitution with `{field}`, notice the curly brackets there. For

Reviewer: ```suggestion
Besides using a predefined value, you can also use a substitution with `{field}`, notice the curly brackets there. For
```
